RecoPixelVertexing/PixelVertexFinding/plugins/gpuFitVertices.h

```diff
@@ -12,28 +12,32 @@
 
 namespace gpuVertexFinder {
 
-  __device__ __forceinline__ void fitVertices(ZVertices* pdata,
-                                              WorkSpace* pws,
+  __device__ __forceinline__ void fitVertices(VtxSoAView pdata,
+                                              WsSoAView pws,
                                               float chi2Max  // for outlier rejection
   ) {
     constexpr bool verbose = false;  // in principle the compiler should optmize out if false
 
-    auto& __restrict__ data = *pdata;
-    auto& __restrict__ ws = *pws;
-    auto nt = ws.ntrks;
-    float const* __restrict__ zt = ws.zt;
-    float const* __restrict__ ezt2 = ws.ezt2;
-    float* __restrict__ zv = data.zv;
-    float* __restrict__ wv = data.wv;
-    float* __restrict__ chi2 = data.chi2;
-    uint32_t& nvFinal = data.nvFinal;
-    uint32_t& nvIntermediate = ws.nvIntermediate;
-
-    int32_t* __restrict__ nn = data.ndof;
-    int32_t* __restrict__ iv = ws.iv;
-
-    assert(pdata);
+    auto& __restrict__ data = pdata;
+    auto& __restrict__ ws = pws;
+    auto nt = ws.ntrks();
+    float const* __restrict__ zt = ws.zt();
+    float const* __restrict__ ezt2 = ws.ezt2();
+    float* __restrict__ zv = data.zv();
+    float* __restrict__ wv = data.wv();
+    float* __restrict__ chi2 = data.chi2();
+    uint32_t& nvFinal = data.nvFinal();
+    uint32_t& nvIntermediate = ws.nvIntermediate();
+
+    int32_t* __restrict__ nn = data.ndof();
+    int32_t* __restrict__ iv = ws.iv();
+
     assert(zt);
+    assert(ezt2);
+    assert(zv);
+    assert(wv);
+    assert(chi2);
+    assert(nn);
```

Comment:
How useful are the asserts on the non-nullness of the View member columns? I'd think the View should either be fully null (default-constructed?) or fully non-null, so if any check is necessary it should be on the View itself.

Reply:
We discussed with @fwyzard and, even though he suggested it's not completely necessary, that the next best thing to do, from within the kernel. 

Asserting the "nullness" of the whole buffer is not possible from within the kernel, AFAIK, and should be done before launching the kernel.

Reply:
We can try to make sure a `View` can never have `nullptr` arguments, which would make these checks redundant.
I think you can remove them already now, or if you prefer to keep them add a comment saying
```c++
    // TODO remove the asserts once the validity of a View is guaranteed at construction time
```